lib/knetwork/dns.go

```diff
@@ -0,0 +1,191 @@
+package knetwork
+
+import (
+	"errors"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/logger/klog"
+	"github.com/miekg/dns"
+	"net"
+	"strconv"
+	"sync"
+)
+
+var (
+	DNSEntryNotExistError = errors.New("the following entry did not exist")
+	DNSEntryExistError    = errors.New("the following entry did exist")
+)
+
+func NewDNS(mods ...DNSModifier) (*DnsServer, error) {
+	defaultLogger, err := klog.New("default", klog.FromFlags(*klog.DefaultFlags()))
```

Comment:
most libraries in the repo don't create a logger. They default to `&NilLogger`, and expect the caller to provide one.
I recommend to follow the same pattern. To all effects, the caller will have a logger and pass it in.

Reply:
Defaults to NilLogger now